src/timed.rs

```diff
@@ -111,50 +111,52 @@ pub struct Timed<A: Actor> {
     queue: BinaryHeap<QueueItem<A::Message>>,
 }
 
-impl<M: Send + 'static, A: Actor<Context = TimedContext<M>, Message = M>> Timed<A> {
+impl<M: Send + 'static, A: Actor<Context = TimedContext<M>, Message = M>> Timed<A>
+where
+    <A as Actor>::Error: From<SendError>,
+{
     pub fn new(inner: A) -> Self {
         Self { inner, queue: Default::default() }
     }
 
     /// Process any pending messages in the internal queue, calling wrapped actor's `handle()`.
-    fn process_queue(&mut self, context: &mut <Self as Actor>::Context) -> Result<(), A::Error> {
-        // Handle all messages that should have been handled by now.
-        let now = Instant::now();
-        while self.queue.peek().map(|m| m.fire_at <= now).unwrap_or(false) {
+    fn process_queue(&mut self, context: &mut <Self as Actor>::Context) -> Result<(), SendError> {
+        // If the message on top of the queue is due, send it to the regular actor queue.
+        // No problem if there are multiple such messages, it's handle() will call process_queue()
+        // again.
```

Comment:
It's not clear to me how 

> it's handle() will call process_queue() again.

As far as I see, the inner actor doesn't have any knowledge of being wrapped in `Timed` and thus can't call outer's methods. 

EDIT: I now see, how this is meant. We're sending the due message to the outer as `Instant` effectively calling outer's `handle` which indeed then calls `process_queue()`.

Make me think if this wording would be clearer:

```suggestion
        // If the message on top of the queue is due, send it to ourselves as `Instant` to enqueue it in the regular actor queue.
        // No problem if there are multiple such messages, next handle() will call process_queue()
        // again.
```

Reply:
@goodhoko thanks for the suggestion, also using (with tiny wordsmith)